File: ppc/lets-play-osumania/solve.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let cnt = 0, n = 0, lines = ['    '.split('')];
rl.on('line', line => {
    if (n === 0) {
        n = parseInt(line);
    } else {
        lines.push(line.trim().slice(1, 5).split(''));
    }
    if(lines.length === n + 1) {
        lines.push('    '.split(''));
        for (let j = 0; j < 4; j ++) {
            for (let i = 1; i <= n; i++) {
                if (lines[i][j] === '-' && [' ', '-', '#'].indexOf(lines[i - 1][j]) !== -1 && [' ', '-'].indexOf(lines[i + 1][j]) !== -1) {
                   cnt += 1;
                }
            }
        }
        console.log(cnt);
        process.exit(0);
    }
})

File: ppc/lets-play-osumania/solve.py
n = int(input())
cnt, lines = 0, [[' '] * 4]
lines.extend([list(input()[1:-1]) for _ in range(n)])
lines.append([' '] * 4)
for j in range(4):
    for i in range(1, n + 1):
        if lines[i][j] == '-' and lines[i - 1][j] in [' ', '-', '#'] and lines[i + 1][j] in [' ', '-']:
            cnt += 1
print(cnt)